tls/s2n_record_write.c

```diff
@@ -82,6 +82,21 @@ S2N_RESULT s2n_record_max_write_payload_size(struct s2n_connection *conn, uint16
     return S2N_RESULT_OK;
 }
 
+S2N_RESULT s2n_record_max_write_record_size(struct s2n_connection *conn, uint16_t max_fragment_size, uint16_t *max_size)
```

Comment:
Sort of. I'm not sure I'd say "fragment length" instead of "max fragment length" though, because it's based on the local max fragment length rather than the size of the data about to be written in the current record. There's a maximum value the "max fragment length" can be (2^14), but locally we can also enforce a lower "max fragment length" (like the default of 8K).

And idk how I ended up with a name that included "record" twice, good catch xD